### PR TITLE
boards: mm_swiftio: support west command

### PR DESCRIPTION
**Pull request by lgl88911**
Add burner for pyocd to support:
west flash
west debug
west debugserver

Signed-off-by: Frank Li <lgl88911@163.com>

### CODE REVIEW

File: boards/arm/mm_swiftio/burner/pyocd_user.py [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+# SPDX-License-Identifier: Apache-2.0
+
+import os
+
+def will_connect():
+    # Look up the external flash memory region.
+    extFlash = target.memory_map.get_region_by_name("flexspi")
+
+    # Set the path to an .FLM flash algorithm.
+    flm_path = os.getenv('ZEPHYR_BASE') + '/boards/arm/mm_swiftio/burner/MIMXRT105x_QuadSPI_4KB_SEC.FLM'
```
Comment (mbolivar) [inline boards/arm/mm_swiftio/burner/pyocd_user.py]: Could just use `pathlib.Path(__file__).parent / 'MIMXRT105x_QuadSPI_4KB_SEC.FLM'` to keep this self-contained.

File: boards/arm/mm_swiftio/burner/pyocd_user.py [hunk 2]
```diff
@@ -0,0 +1,13 @@
+# Copyright (c) 2020, MADMACHINE LIMITED
+#
+# SPDX-License-Identifier: Apache-2.0
+
+import os
+
+def will_connect():
+    # Look up the external flash memory region.
+    extFlash = target.memory_map.get_region_by_name("flexspi")
```
Comment (mbolivar) [inline boards/arm/mm_swiftio/burner/pyocd_user.py]: Does adding 'global target' before this line fix your pylint issue?

File: boards/arm/mm_swiftio/burner/pyocd_user.py (hunk 1 above)
Comment (lgl88911) [reply boards/arm/mm_swiftio/burner/pyocd_user.py]: Thanks so much. I fixed it.  I'm a beginner in python and don't know how to deal with this problem, now you provide the answer to me。 I'll close https://lists.zephyrproject.org/g/devel/topic/add_pyocd_user_scripy_cause/71021588?p=,,,20,0,0,0::recentpostdate%2Fsticky,,,20,2,0,71021588

File: boards/arm/mm_swiftio/burner/pyocd_user.py (hunk 2 above)
Comment (lgl88911) [reply boards/arm/mm_swiftio/burner/pyocd_user.py]: Thanks so much. I fixed it. I'm a beginner in python and don't know how to deal with this problem, now you provide the answer to me。 I'll close https://lists.zephyrproject.org/g/devel/topic/add_pyocd_user_scripy_cause/71021588?p=,,,20,0,0,0::recentpostdate%2Fsticky,,,20,2,0,71021588

File: boards/arm/mm_swiftio/burner/pyocd_user.py (hunk 1 above)
Comment (lgl88911) [reply boards/arm/mm_swiftio/burner/pyocd_user.py]: using pathlib is a good idea, I test __file__ in other file is OK, below is test code.
```
import pathlib
import platform

print(platform.python_version())
print(pathlib.Path(__file__).parent)
```
But the user script in pyocd can't access global __file__, below is modify code
```
import pathlib

def will_connect():
    # Look up the external flash memory region.
    global target
    extFlash = target.memory_map.get_region_by_name("flexspi")

    # Set the path to an .FLM flash algorithm.
    flm_path = pathlib.Path(__file__).parent / 'MIMXRT105x_QuadSPI_4KB_SEC.FLM'
    extFlash.flm = flm_path
```
When I do west flash will report can't find __file__
```
0000612:CRITICAL:__main__:uncaught exception: name '__file__' is not defined
Traceback (most recent call last):
  File "/home/frank/.local/lib/python3.6/site-packages/pyocd/__main__.py", line 343, in run
    self._COMMANDS[self._args.cmd](self)
  File "/home/frank/.local/lib/python3.6/site-packages/pyocd/__main__.py", line 473, in do_flash
    with session:
  File "/home/frank/.local/lib/python3.6/site-packages/pyocd/core/session.py", line 287, in __enter__
    self.open()
  File "/home/frank/.local/lib/python3.6/site-packages/pyocd/core/session.py", line 367, in open
    self._board.init()
  File "/home/frank/.local/lib/python3.6/site-packages/pyocd/board/board.py", line 80, in init
    self.delegate.will_connect(board=self)
  File "/home/frank/.local/lib/python3.6/site-packages/pyocd/core/session.py", line 412, in __call__
    self._fn(**args)
  File "<string>", line 13, in will_connect
NameError: name '__file__' is not defined
```
Do you have any other suggestions for these? Thanks

**Review by MaureenHelm — CHANGES_REQUESTED** [review]
The project hasn't decided yet if/how to accept binary blobs, so the .FLM file is a problem.

**Comment by jfischer-phytec-iot** [comment]
> The project hasn't decided yet if/how to accept binary blobs, so the .FLM file is a problem.

Flash algorithm could be build from algorithm source locally, no need for this binary. BTW, BIN +1.44 MB for flash algorithm?

File: boards/arm/mm_swiftio/burner/pyocd_user.py (hunk 1 above)
Comment (mbolivar) [reply boards/arm/mm_swiftio/burner/pyocd_user.py]: That's very strange. Please feel free to disregard this comment, then. Thanks.

File: boards/arm/mm_swiftio/burner/pyocd_user.py (hunk 2 above)
Comment (mbolivar) [reply boards/arm/mm_swiftio/burner/pyocd_user.py]: It looks like you can do something like this:

```
        # 'target' is global
        # pylint: disable=undefined-variable
```

See here for an example: https://github.com/zephyrproject-rtos/ci-tools/pull/128/files

**Comment by lgl88911** [comment]
> > The project hasn't decided yet if/how to accept binary blobs, so the .FLM file is a problem.
> 
> Flash algorithm could be build from algorithm source locally, no need for this binary. BTW, BIN +1.44 MB for flash algorithm?

Yes FLW file is a flash algorithm, The BIN file come from MDK Keil, It' will be load by pyocd to write flash.  
If the zephyr project does not accept this method, maybe I have to research pyocd and submit the correct code to pyocd to solve the problem of using west flash on mm_swiftio, or have other suggestions to deal with this problem?

File: boards/arm/mm_swiftio/burner/pyocd_user.py (hunk 2 above)
Comment (lgl88911) [reply boards/arm/mm_swiftio/burner/pyocd_user.py]: Thanks so mush, It be fixed

**Review by mbolivar — COMMENTED** [review]
The west related parts of this look good. I'm going to remove myself from the reviewers list so the pull reminders robot leaves me alone while you're deciding what to do about the binary.

**Review by mbolivar — APPROVED** [review]

**Comment by lgl88911** [comment]
https://github.com/mbedmicro/pyOCD/pull/845 fix rt1052 qspi flash algo, So no need load flm by script,  Close this PR and pull https://github.com/zephyrproject-rtos/zephyr/pull/24765

**Comment by lgl88911** [comment]
> The west related parts of this look good. I'm going to remove myself from the reviewers list so the pull reminders robot leaves me alone while you're deciding what to do about the binary.

Thanks for your help, the flash algo algo issue be fixed in pyOCD, so close this pr.

**Pull request closed without merging**